scripts/installer/curl_install_pypi/install.py

```diff
@@ -0,0 +1,360 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+#
+# This script will install the promptflow into a directory and create an executable
+# at a specified file path that is the entry point into the promptflow.
+#
+# The latest versions of all promptflow command packages will be installed.
+#
+
+import os
+import sys
+import platform
+import stat
+import tempfile
+import shutil
+import subprocess
+import hashlib
+
+
+PF_DISPATCH_TEMPLATE = """#!/usr/bin/env bash
+export PF_INSTALLER=Script
+{install_dir}/bin/python -m promptflow._cli._pf.entry "$@"
+"""
+
+PFAZURE_DISPATCH_TEMPLATE = """#!/usr/bin/env bash
+{install_dir}/bin/python -m promptflow._cli._pf_azure.entry "$@"
+"""
+
+PFS_DISPATCH_TEMPLATE = """#!/usr/bin/env bash
+{install_dir}/bin/python -m promptflow._sdk._service.entry "$@"
+"""
+
+DEFAULT_INSTALL_DIR = os.path.expanduser(os.path.join('~', 'lib', 'promptflow'))
+DEFAULT_EXEC_DIR = os.path.expanduser(os.path.join('~', 'bin'))
+PF_EXECUTABLE_NAME = 'pf'
+PFAZURE_EXECUTABLE_NAME = 'pfazure'
+PFS_EXECUTABLE_NAME = 'pfs'
+
+
+USER_BASH_RC = os.path.expanduser(os.path.join('~', '.bashrc'))
+USER_BASH_PROFILE = os.path.expanduser(os.path.join('~', '.bash_profile'))
+
+
+class CLIInstallError(Exception):
+    pass
+
+
+def print_status(msg=''):
+    print('-- '+msg)
+
+
+def prompt_input(msg):
+    return input('\n===> '+msg)
+
+
+def prompt_input_with_default(msg, default):
+    if default:
+        return prompt_input("{} (leave blank to use '{}'): ".format(msg, default)) or default
+    else:
+        return prompt_input('{}: '.format(msg))
+
+
+def prompt_y_n(msg, default=None):
+    if default not in [None, 'y', 'n']:
+        raise ValueError("Valid values for default are 'y', 'n' or None")
+    y = 'Y' if default == 'y' else 'y'
+    n = 'N' if default == 'n' else 'n'
+    while True:
+        ans = prompt_input('{} ({}/{}): '.format(msg, y, n))
+        if ans.lower() == n.lower():
+            return False
+        if ans.lower() == y.lower():
+            return True
+        if default and not ans:
+            return default == y.lower()
+
+
+def exec_command(command_list, cwd=None, env=None):
+    print_status('Executing: '+str(command_list))
+    subprocess.check_call(command_list, cwd=cwd, env=env)
+
+
+def create_tmp_dir():
+    tmp_dir = tempfile.mkdtemp()
+    return tmp_dir
+
+
+def create_dir(dir):
+    if not os.path.isdir(dir):
+        print_status("Creating directory '{}'.".format(dir))
+        os.makedirs(dir)
+
+
+def is_valid_sha256sum(a_file, expected_sum):
+    sha256 = hashlib.sha256()
+    with open(a_file, 'rb') as f:
+        sha256.update(f.read())
+    computed_hash = sha256.hexdigest()
+    return expected_sum == computed_hash
+
+
+def create_virtualenv(install_dir):
+    cmd = [sys.executable, '-m', 'venv', install_dir]
```

Comment:
what happened if we run multiple times to same dir?

Reply:
have replied in this chat
![image](https://github.com/microsoft/promptflow/assets/26239730/76665f97-b7fe-495c-9651-0ab2132ede2b)